### PR TITLE
Application and Access log - initial version

### PR DESCRIPTION
**Pull request by aryszka**

### CODE REVIEW

**Comment by aryszka** [comment]
created this to the wip branch, only to avoid double work

File: log/access.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+	ff := r.Header.Get("X-Forwarded-For")
+	if ff != "" {
+		return ff
+	}
+
+	return r.RemoteAddr
+}
+
+// strip port from addresses with hostname, ipv4 or ipv6
+func stripPort(address string) string {
```
Comment (lmineiro) [inline log/access.go]: https://golang.org/pkg/net/#SplitHostPort
Recently revived my ExtractDomain function using that one also

Comment (aryszka) [inline log/access.go]: updated

File: log/access.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+	h, _, err := net.SplitHostPort(address)
+	if err != nil {
+		h = ""
```
Comment (lmineiro) [inline log/access.go]: Are you sure this is what you want to do? If address doesn't have a port it may still have a valid address...

**Pull request merged**